src/engine/pieces/queen.ts

```diff
@@ -1,13 +1,18 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import { getDiagonalMoves, getLateralMoves } from '../helper';
 
 export default class Queen extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        const currentPosition = board.findPiece(this);
+        const lateralMoves = getLateralMoves(board, currentPosition);
+        const diagonalMoves = getDiagonalMoves(board, currentPosition);
+
+        return lateralMoves.concat(diagonalMoves);
```

Comment:
```suggestion
        return [...getLateralMoves(board, currentPosition), ...getDiagonalMoves(board, currentPosition)]
```
You could simplify this by using the [spread operator](https://www.w3schools.com/react/react_es6_spread.asp) which allows you to combine all or part of objects such a arrays together easily